docs/asdf/developer_overview.rst

```diff
@@ -0,0 +1,350 @@
+High level overview of the basic ASDF library
+=============================================
+
+This document is an attempt to make it easier to understand the design and
+workings of the python asdf library for those unfamiliar with it. This is
+expected to grow organically so at the moment it should not be considered
+complete or comprehensive.
+
+The design is somewhat complicated by the fact that in using pyyaml and
```

Comment:
This is already scary :) I just started reading this document and already discovered that I was supposed to know `pyyaml` and `jsonschema` and that these packages were used in `asdf`. I think reformulating this might help.

Also ` in using ..., one uses them by supplying code for these libraries to use` This is very confusing formulation for me and verb "use" used three times in a single sentence.

Reply:
@mcara Not sure what you mean by the first (regarding reformulating). As to the second, I blame a trancelike mode resulting in stream of gibberish text.

Reply:
_The design is somewhat complicated by the fact that in using pyyaml..."_ sounds to me like I am suddenly presented with this new package `pyyaml`. Maybe the sentence should be changed to (if one assumes that `pyyaml` is a prerequisite) something that introduces `pyyaml` in a more smooth way. Also, if it is a prerequisite, then state so at the beginning. If not - introduce it first.

Another thing that is not clear to me: _"The design is somewhat..."_ Design of what? Of schemas, tags, or _this_ document?

Reply:
I've try to write this more clearly in the version I just pushed.